components/director/internal/domain/oauth20/service.go

```diff
@@ -0,0 +1,175 @@
+package oauth20
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const clientCredentialScopesPrefix = "__clientCredentialsRegistrationScopes"
+const applicationJSONType = "application/json"
+
+var defaultGrantTypes = []string{"client_credentials"}
+
+//go:generate mockery -name=ScopeCfgProvider -output=automock -outpkg=automock -case=underscore
+type ScopeCfgProvider interface {
+	GetRequiredScopes(path string) ([]string, error)
+}
+
+//go:generate mockery -name=UIDService -output=automock -outpkg=automock -case=underscore
+type UIDService interface {
+	Generate() string
+}
+
+type service struct {
+	clientEndpoint            string
+	publicAccessTokenEndpoint string
+	scopeCfgProvider          ScopeCfgProvider
+	httpCli                   *http.Client
```

Comment:
What do you think about extracting logic with calls to seperate object for example: `HydraCLI`?
It will be easier to test and cleaner.
I would treat this something like repository.

Reply:
I don't see a need for that, as the service is purely all about calling Hydra, right?

Reply:
Thanks, so I mark it as resolved